operations/src/main/java/edu/unc/lib/boxc/operations/impl/download/DownloadBulkService.java

```diff
@@ -0,0 +1,111 @@
+package edu.unc.lib.boxc.operations.impl.download;
+
+import edu.unc.lib.boxc.auth.api.Permission;
+import edu.unc.lib.boxc.auth.api.models.AccessGroupSet;
+import edu.unc.lib.boxc.auth.api.services.AccessControlService;
+import edu.unc.lib.boxc.model.api.objects.ContentObject;
+import edu.unc.lib.boxc.model.api.objects.FileObject;
+import edu.unc.lib.boxc.model.api.objects.RepositoryObjectLoader;
+import edu.unc.lib.boxc.model.api.objects.WorkObject;
+import edu.unc.lib.boxc.model.fcrepo.ids.PIDs;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * Service that manages downloading all of a Work's FileObjects as a zip file
+ */
+public class DownloadBulkService {
+    private AccessControlService aclService;
+    private RepositoryObjectLoader repoObjLoader;
+    private String basePath;
+
+    public String downloadBulk(DownloadBulkRequest request) throws IOException {
+        var workPid = PIDs.get(request.getWorkPidString());
+        var agentPrincipals = request.getAgent().getPrincipals();
+        aclService.assertHasAccess(
+                "User does not have permissions to view the Work for download",
+                workPid, agentPrincipals, Permission.viewOriginal);
+
+        var workObject = repoObjLoader.getWorkObject(workPid);
+        if (workObject == null) {
+            throw new IllegalArgumentException("Failed to bulk download for " + workPid);
+        }
+
+        var zipFilePath = basePath + getZipFilename(workPid.getId());
+
+        zipFiles(workObject, agentPrincipals, zipFilePath);
+        return zipFilePath;
+    }
+
+    private void zipFiles(WorkObject workObject, AccessGroupSet agentPrincipals, String zipFilePath) throws IOException {
+        var memberObjects = workObject.getMembers();
+        final FileOutputStream fos = new FileOutputStream(zipFilePath);
```

Comment:
Just a note that if we pass zipFilePath in as a path, we can initialize a FileOutputStream directly with a Path using `Files.newOutputStream(zipFilePath)`